### ISSUE TITLE
Crop the face area of the images when flipped.

### ISSUE THREAD

**Issue by datbu178** [eladrich/pixel2style2pixel#99]
Thank you for the great project. I am very impressed with your work!  I am trying to rewrite your code using Tensorflow and I discovered a problem. You crop out the face area of the image before entering the Face Recognition Model with the following lines of code: 
x = x[:, :, 35:223, 32:220]
However, this cropping area is not symmetrical to the left and right of the image. So with the same image, when flipping and then cropping the face area, will give a different face area (compared to when not flipping the image).
So would it be better if we adjusted the cropped area to the same when flipping the image?  Thank you!

**Comment by yuval-alaluf** [comment]
Hi @datbu178 -- good observation. 
While the face cropping we use is not symmetrical, it was not exactly designed to be symmetrical. Instead, the cropping is supposed to simulate the cropping obtained with MTCNN or some other face detection model without needing to explicitly pass the image to the detector (thereby saving a lot of time during training). 
At the end of the day, the cropping is used to measure the identity similarity so I am not sure how much the identity score would change if we made the cropping more exact. 
Therefore, to answer your question, I am not sure how adjusting the cropped area when flipping the image will be influence the results. Also, it seems a bit non-trivial to keep track of which images were flipped and which weren't when computing the loss.
You're welcome to try and let us know if you find anything interesting 💪

**Comment by datbu178** [comment]
Thank you! I will try it.

**Comment by LiUzHiAn** [comment]
Hi, may I know how did you get the coordinates to exact cropped regions? i.e., the `[:,:,35:222,32:220]`.

In my perspective, If the input images are 1024 resolution, in this case, only part of the face is cropped?

**Comment by datbu178** [comment]
Hi @LiUzHiAn .

> In my perspective, If an input image is 1024 resolution, in this case, only part of the face is cropped?

The image will be resized to 256x256 before being cropped. So the cropped image will include the entire face and part of the surrounding landscape.
Each image has a different face area, and I think [:,:,35:222,32:220] would be a good approximation to fit all images. If you want this area to be smaller, you can adjust it yourself.

**Comment by LiUzHiAn** [comment]
@datbu178 

 Yes, I got it.  Thank you so much.
